File: backend/src/Types/Dto.fs
namespace Bank

type CreateBankTransferDto = {
  FromAccountId: int64 option
  ToAccountId: int64 option
  AmountEurCents: int64 option
}

type CreateBankTransferResponseDto = {
  Transfer: BankTransfer
}

type GetAccountOwnersResponseDto = {
  Owners: AccountOwner list
}

type GetBankAccountsResponseDto = {
  Accounts: BankAccount list
}

File: frontend/src/types/dto.ts
import { AccountOwner, BankAccount, BankTransfer } from './domain'

type ResponseCommon = {
  loading: boolean
  error: boolean
  message?: string
}

type GetOwnersResponse = ResponseCommon & { owners?: AccountOwner[] }
type GetAccountsResponse = ResponseCommon & { accounts?: BankAccount[] }
type GetTransfersResponse = ResponseCommon & { transfers?: BankTransfer[] }

type CreateTransferRequest = {
  fromAccountId: number
  toAccountId: number
  amountEurCents: number
}

type CreateTransferResponse = ResponseCommon & { transfer?: BankTransfer }

export type {
  GetOwnersResponse,
  GetAccountsResponse,
  GetTransfersResponse,
  CreateTransferRequest,
  CreateTransferResponse,
}
